iree/compiler/Dialect/Flow/Transforms/test/hlo_to_hlo_preprocessing.mlir

```diff
@@ -151,3 +151,51 @@ func @reorder_broadcast_in_dim_2d(%arg0: tensor<2x4xi32>, %arg1: tensor<2x4xi32>
   // CHECK: return %[[BCAST]]
   return %2 : tensor<3x2x4xi32>
 }
+
+// -----
+
+// CHECK: @reorder_broadcast_in_dim_scalar_unary(%[[ARG0:.*]]: tensor<f32>) -> tensor<1x8x8x64xf32>
+func @reorder_broadcast_in_dim_scalar_unary(%arg0: tensor<f32>) -> tensor<1x8x8x64xf32> {
+  // CHECK: %[[EXP:.*]] = "mhlo.exponential"(%[[ARG0]]) : (tensor<f32>) -> tensor<f32>
+  // CHECK: %[[BCAST:.*]] = "mhlo.broadcast_in_dim"(%[[EXP]]) {broadcast_dimensions = dense<> : tensor<0xi64>} : (tensor<f32>) -> tensor<1x8x8x64xf32>
+  %0 = "mhlo.broadcast_in_dim"(%arg0) {broadcast_dimensions = dense<[]> : tensor<0xi64>} : (tensor<f32>) -> tensor<1x8x8x64xf32>
+  %1 = "mhlo.exponential"(%0) : (tensor<1x8x8x64xf32>) -> tensor<1x8x8x64xf32>
```

Comment:
Can we add other cases like

```
%2 = "mhlo.abs" (%0) ...
%3 = "mhlo.ceil" (%0) ...
```

So we can cover all the cases.

I think it works even the values are unused (ie we dont need to return them) because there is no cse pass in the test.

Reply:
Umm, seems that it's got removed in the test. Is dead code elimination enabled for iree-opt by default?

Reply:
hmm, I'm not sure on this part. Having multiple functions is also good to me.

Reply:
Updated. See if this is better :P.